src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -0,0 +1,87 @@
+import { BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ChangeEvent, FC, useState } from "react";
+
+import useAmountInputtStyles from "src/views/home/components/amount-input/amount-input.styles";
+import Typography from "src/views/shared/typography/typography.view";
+import { Token } from "src/domain";
+
+interface onChangeParams {
+  amount: BigNumber;
+  isInvalid: boolean;
+}
+
+interface AmountInputProps {
+  token: Token;
+  balance: BigNumber;
+  fee: BigNumber;
+  onChange: (params: onChangeParams) => void;
+}
+
+const AmountInput: FC<AmountInputProps> = ({ token, balance, fee, onChange }) => {
+  const [value, setValue] = useState("");
+  const classes = useAmountInputtStyles(value.length);
+  const actualFee = token.symbol.includes("ETH") ? fee : BigNumber.from(0);
+
+  const getFixedTokenAmount = (amount: string, decimals = 18): string => {
+    const amountWithDecimals = Number(amount) / Math.pow(10, decimals);
+    return Number(amountWithDecimals.toFixed(decimals)).toString();
+  };
+  const updateAmountInput = (amount: BigNumber) => {
+    const newAmountWithFee = amount.add(actualFee);
+    const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(BigNumber.from(balance));
+    const areFundsExceededDueToFee =
+      isNewAmountWithFeeMoreThanFunds && amount.lte(BigNumber.from(balance));
+    const isAmountInvalid =
+      isNewAmountWithFeeMoreThanFunds || areFundsExceededDueToFee || amount.isZero();
+
+    onChange({
+      amount,
+      isInvalid: isAmountInvalid,
+    });
+  };
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const decimals = token.decimals;
+    const regexToken = `^(?!0\\d)\\d*(?:\\.\\d{0,${decimals}})?$`;
+    const INPUT_REGEX = new RegExp(regexToken);
+    if (event.target.value === "") {
```

Comment:
We don't need to check for `""` neither for `"."` here, the regex already take care of this. I tried the original code and it works as expected:

```ts
if (INPUT_REGEX.test(event.target.value)) {
  const tokensValue = event.target.value.length > 0 ? event.target.value : "0";
  const newAmountInTokens = parseUnits(tokensValue, token.decimals);

  setValue(event.target.value);
  updateAmountInput(newAmountInTokens);
}
```

Reply:
I want to keep `"."`because allow to start with a dot to type `0.`

Reply:
With the previous code you are already allowed to do that